### PR TITLE
Changes related to fixing the inheritance logic.

### PR DESCRIPTION
**Pull request by sarangan12**
As discussed offline, this PR is related to https://github.com/Azure/autorest.ruby/pull/13 and fixes the tests associated with those changes. 

@vishrutshah @veronicagg @salameer Please review and approve

### CODE REVIEW

File: management/azure_mgmt_compute/spec/virtual_machines_spec.rb [hunk 1]
```diff
@@ -113,11 +113,11 @@
     ni = network_client.make_request(:get, ni_path, options)
     expect(ni).not_to be_nil
     # user NetworkInterface mapper to deserialize object
-    ni_instance  = NetworkInterface.new.from_json(ni)
+    ni_instance  = network_client.deserialize(NetworkInterface.mapper(), ni)
     # retrieve first IP address for first IP configuration
     ip_address_path = ni_instance.ip_configurations[0].public_ipaddress.id
     ip_address = network_client.make_request(:get, ip_address_path, options)
-    ip_address_instance = PublicIPAddress.new.from_json(ip_address)
+    ip_address_instance = network_client.deserialize(PublicIPAddress.mapper(), ip_address)
```
Comment (vishrutshah) [inline management/azure_mgmt_compute/spec/virtual_machines_spec.rb]: just curious what has caused this change?

File: management/azure_mgmt_service_bus/spec/queues_spec.rb [hunk 2]
```diff
@@ -21,7 +21,6 @@
     namespace_parameters = Azure::ARM::ServiceBus::Models::SBNamespace.new
     namespace_parameters.location = 'westus'
     queue_parameters = Azure::ARM::ServiceBus::Models::SBQueue.new
-    queue_parameters.location = 'westus'
```
Comment (vishrutshah) [inline management/azure_mgmt_service_bus/spec/queues_spec.rb]: any reason for removing this? Just wanted to make sure what has caused this?

File: management/azure_mgmt_compute/spec/virtual_machines_spec.rb (hunk 1 above)
Comment (sarangan12) [reply management/azure_mgmt_compute/spec/virtual_machines_spec.rb]: I was working with David on a different task. It was dropped eventually. This was a remnant from that task

File: management/azure_mgmt_service_bus/spec/queues_spec.rb (hunk 2 above)
Comment (sarangan12) [reply management/azure_mgmt_service_bus/spec/queues_spec.rb]: Because accroding to the Resource definition in ServiceBus queue, there is no location. https://github.com/Azure/azure-rest-api-specs/blob/current/specification/servicebus/resource-manager/Microsoft.ServiceBus/2017-04-01/servicebus.json#L2503

**Comment by sarangan12** [comment]
@veronicagg The CI will pass as it is running against the current code.

**Comment by sarangan12** [comment]
@vishrutshah Changed the expect per offline discussion

File: management/azure_mgmt_compute/spec/virtual_machines_spec.rb [hunk 3]
```diff
@@ -135,7 +135,7 @@
 
     result = @resource_helper.network_client.network_interfaces.list_all
     result.each do |network_interface|
-      expect(network_interface.virtual_machine).to be_an_instance_of(MsRestAzure::SubResource)
+      expect(network_interface.virtual_machine).to be_an_instance_of(VirtualMachine)
```
Comment (vishrutshah) [inline management/azure_mgmt_compute/spec/virtual_machines_spec.rb]: you should have to write `Azure::ARM::Compute::Models::VirtualMachine`

Comment (sarangan12) [reply management/azure_mgmt_compute/spec/virtual_machines_spec.rb]: Actually, it should be Azure::ARM::Network::Models::SubResource. The model looks like this:

```
class NetworkInterface < Resource

      include MsRestAzure

      # @return [SubResource] The reference of a virtual machine.
      attr_accessor :virtual_machine
```

So, changed it with the latest commit

File: management/azure_mgmt_analysis_services/lib/generated/azure_mgmt_analysis_services/models/analysis_services_server_update_parameters.rb [hunk 4]
```diff
@@ -12,7 +12,6 @@ class AnalysisServicesServerUpdateParameters
 
       include MsRestAzure
 
-      include MsRest::JSONable
```
Comment (veronicagg) [inline management/azure_mgmt_analysis_services/lib/generated/azure_mgmt_analysis_services/models/analysis_services_server_update_parameters.rb]: why was this included before?

Comment (sarangan12) [reply management/azure_mgmt_analysis_services/lib/generated/azure_mgmt_analysis_services/models/analysis_services_server_update_parameters.rb]: It was a remnant from a serialization task which was later dropped.

**Comment by sarangan12** [comment]
@veronicagg Added the link in the comment.
